src/Shipment/Request/PostShipmentsRequest.php

```diff
@@ -106,44 +105,27 @@ protected function encodeShipment(Shipment $shipment): array
     }
 
     /**
-     * @param  \MyParcelNL\Pdk\Base\Support\Collection $groupedShipments
+     * @param  \MyParcelNL\Pdk\Shipment\Model\Shipment $shipment
      *
-     * @return null|array
+     * @return array
+     * @throws \MyParcelNL\Pdk\Base\Exception\InvalidCastException
      */
-    private function encodeSecondaryShipments(Collection $groupedShipments): ?array
+    private function encodeShipmentWithSecondaryShipments(Shipment $shipment): array
     {
-        $clonedCollection = new Collection($groupedShipments->all());
-
-        $clonedCollection->shift();
+        $encodedShipment = $this->encodeShipment($shipment);
+        $schema          = Pdk::get(CarrierSchema::class);
 
-        if ($clonedCollection->isEmpty()) {
-            return null;
-        }
+        $schema->setCarrier($shipment->deliveryOptions->carrier);
 
-        /**
-         * Turn grouped shipments into regular collection to avoid all shipment properties being added.
-         */
-        return $clonedCollection
-            ->map(function (Shipment $shipment) {
-                return [
-                    'reference_identifier' => $shipment->externalIdentifier,
-                ];
-            })
-            ->toArray();
-    }
+        $secondaryShipmentsAmount = $schema->canHaveMultiCollo() ? $shipment->deliveryOptions->labelAmount - 1 : 0;
 
-    /**
-     * @param  \MyParcelNL\Pdk\Base\Support\Collection $groupedShipments
-     *
-     * @return array
-     * @throws \MyParcelNL\Pdk\Base\Exception\InvalidCastException
-     */
-    private function encodeShipmentWithSecondaryShipments(Collection $groupedShipments): array
-    {
-        $shipment                        = $this->encodeShipment($groupedShipments->first());
-        $shipment['secondary_shipments'] = $this->encodeSecondaryShipments($groupedShipments);
+        for ($i = 0; $i < $secondaryShipmentsAmount; $i++) {
+            $encodedShipment['secondary_shipments'][] = [
```

Comment:
liever dit netjes in een array doen en dan met array_replace oid samengevoegd returnen, leest makkelijker